src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@SubJunk I agree that if the ternaries get too complex it's preferrable to switch to ```if...else``` instead, but in this case it's actually very basic and I find terniaries *more* readable when the complexity is low (not to mention that the can save a lot of hazzle when used inline on method arguments). The only think that makes this line "too long" is the descriptive identifiers combined with the level of indentation. Those things also change while you work with the code, so it might have fitted just fine when you first wrote it.

This one could still be on one line actually, I think the line length was 119 if it was all on one line. Even though many still treats the 80 characters width as if it were "set in stone", I think it's oulived its usefullness a long time ago. It does stem from the pre-GUI era where a screen was typically 80 characters wide, which isn't really relevant anymore.

I can certainly change this one to ```if...else```, but what I'm really after is how to split too long lines with terniary expressions when that need arise.